lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -663,6 +665,11 @@ data ApiBlockReference = ApiBlockReference
     , absoluteSlotNumber :: !(ApiT SlotNo)
     } deriving (Eq, Generic, Show)
 
+data ApiSlotReference = ApiSlotReference
+    { time :: !UTCTime
+    , absoluteSlotNumber :: !(ApiT SlotNo)
+    } deriving (Eq, Generic, Show)
```

Comment:
What about including also the epoch and slot within epoch to be consistent with other time references? We use:

```hs
data ApiTimeReference = ApiTimeReference
    { time :: !UTCTime
    , block :: !ApiBlockReference
    } deriving (Eq, Generic, Show)

data ApiBlockReference = ApiBlockReference
    { epochNumber :: !(ApiT EpochNo)
    , slotNumber :: !(ApiT SlotInEpoch)
    , height :: !(Quantity "block" Natural)
    , absoluteSlotNumber :: !(ApiT SlotNo)
    } deriving (Eq, Generic, Show)
```

Although indeed, the TTL is about slots and not blocks so the _height_ don't make much sense. The other however do. So I'd suggest to have something along the lines of:

```hs
data ApiTimeReferenceBySlot = ApiTimeReferenceBySlot
    { time :: !UTCTime
    , slot :: !ApiSlotReference
    } deriving (Eq, Generic, Show)

data ApiSlotReference = ApiBlockReference
    { epochNumber :: !(ApiT EpochNo)
    , slotNumber :: !(ApiT SlotInEpoch)
    , absoluteSlotNumber :: !(ApiT SlotNo)
    } deriving (Eq, Generic, Show)
```

Reply:
Not only is TTL about slots, but it's about slots in the _future_ -- which cannot have a block height.

OK I have added epoch and slot numbers.

Reply:
For consistency, It'd be better to have them nested under a `slot` field, don't you think :thinking: ?

Reply:
Yeah maybe - I am going to straighten out the types a bit (without changing existing API fields).